include/openrave/environment.h

```diff
@@ -748,6 +749,9 @@ class OPENRAVE_API EnvironmentBase : public boost::enable_shared_from_this<Envir
     /// \brief update EnvironmentBase according to new EnvironmentBaseInfo
     virtual void UpdateFromInfo(const EnvironmentBaseInfo& info) = 0;
 
+    std::map<std::string, rapidjson::Value> _mExpandedBodyValue;  ///< map of bodyId and expandedBodyValue
```

Comment:
@bucketzxm not sure if it is wise to save `rapidjson::Value` like this, who has control of the allocator?

Reply:
@rdiankov This MR is **not** ready for review.  I haven't decided which is good,  saving a rapidjson::Value or saving an info structure for later serialize.  Also need to address a better way to allocate new document when doing deserialize

Reply:
@bucketzxm ping